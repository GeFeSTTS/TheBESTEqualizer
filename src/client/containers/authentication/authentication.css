.authentication {
    font-family: var(--font-authentication);
    color: var(--authFont-white);
    background: url(../../assets/images/authBackground.jpeg) no-repeat;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
}

.title {
    font-size: 2.8rem;
    text-align: center;
    margin-bottom: 1rem;
}

.form-container {
    width: 320px;
    padding: 0 1rem;
}

.message {
    color: var(--authFont-white);
    background-color: transparent;
    border: none;
    cursor: pointer;
    outline: none;
}

.message:hover {
    text-decoration: underline;
}

@media screen and (min-width: 1440px) {
    .authImage {
        width: 40%;
    }

    .form-container {
        width: 20%;
    }
}

@media screen and (max-width: 1024px) {
    .authImage {
        width: 60%;
    }

    .form-container {
        width: 35%;
    }
}

@media screen and (max-height: 754px) and (max-width: 768px) {
    .authImage {
        display: none;
    }
}

@media screen and (max-width: 768px) {
    .authentication {
        flex-wrap: wrap;
    }
    
    .authImage {
        margin-bottom: -10rem;
    }
    
    .form-container {
        width: 50%;
    }
    
    .title {
        font-size: 2.5rem;
    }
}

@media screen and (max-width: 500px) {
    .authImage {
        display: none;
    }

    .form-container {
        width: 70%;
    }

    .title {
        font-size: 2rem;
    }
}
