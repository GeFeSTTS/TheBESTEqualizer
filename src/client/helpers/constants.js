import Pizzicato from 'pizzicato';

export const HOST = 'http://localhost:8080';

export const fieldsInfo = [
  {
    type: 'text',
    name: 'username',
    label: 'Username',
    isMember: false,
  },
  {
    type: 'text',
    name: 'email',
    label: 'Email',
    isMember: true,
  },
  {
    type: 'password',
    name: 'password',
    label: 'Password',
    isMember: true,
  },
  {
    type: 'password',
    name: 'passwordConfirmation',
    label: 'Pasword confirmation',
    isMember: false,
  },
];

export const JAZZ_PRESET_ARRAY = [
  {
    name: 'Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.Delay({
      feedback: 0,
      time: 0,
      mix: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Ping-Pong Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.PingPongDelay(
      {
        feedback: 0,
        time: 0,
        mix: 0,
      },
    ),
    isVisible: false,
  },
  {
    name: 'Dub Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      cutoff: {
        value: 0,
        minValue: 0,
        maxValue: 4000,
        step: 100,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.DubDelay(
      {
        feedback: 0,
        time: 0,
        cutoff: 0,
        mix: 0,
      },
    ),
    isVisible: false,
  },
  {
    name: 'Distortion',
    effects: {
      gain: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Distortion({
      gain: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Quadrafuzz',
    effects: {
      lowGain: {
        value: 0.18,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      midLowGain: {
        value: 0.06,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      midHighGain: {
        value: 0.2,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      highGain: {
        value: 0.57,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Quadrafuzz({
      lowGain: 0.18,
      midLowGain: 0.06,
      midHighGain: 0.2,
      highGain: 0.57,
      mix: 0,
    }),
    isVisible: true,
  },
  {
    name: 'Flanger',
    effects: {
      time: {
        value: 0.32,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      speed: {
        value: 0.77,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      depth: {
        value: 0.79,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 0.17,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Flanger({
      time: 0.32,
      speed: 0.77,
      depth: 0.79,
      feedback: 0,
      mix: 0.17,
    }),
    isVisible: true,
  },
  {
    name: 'Reverb',
    effects: {
      time: {
        value: 0,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
      decay: {
        value: 0,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Reverb({
      time: 0,
      decay: 0,
      mix: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Tremolo',
    effects: {
      speed: {
        value: 5.86,
        minValue: 0,
        maxValue: 20,
        step: 0.01,
      },
      depth: {
        value: 0.4,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 0.61,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Tremolo({
      speed: 5.86,
      depth: 0.4,
      mix: 0.61,
    }),
    isVisible: true,
  },
  {
    name: 'Stereo panner',
    effects: {
      pan: {
        value: 0,
        minValue: -1,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.StereoPanner({
      pan: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Compressor',
    effects: {
      threshold: {
        value: -21,
        minValue: -100,
        maxValue: 0,
        step: 1,
      },
      knee: {
        value: 8,
        minValue: 0,
        maxValue: 40,
        step: 1,
      },
      attack: {
        value: 0.73,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
      release: {
        value: 0.73,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
      ratio: {
        value: 18,
        minValue: 1,
        maxValue: 20,
        step: 1,
      },
    },
    createEffect: new Pizzicato.Effects.Compressor({
      threshold: -21,
      knee: 8,
      attack: 0.73,
      release: 0.73,
      ratio: 18,
    }),
    isVisible: true,
  },
  {
    name: 'Low-Pass Filter',
    effects: {
      frequency: {
        value: 12308,
        minValue: 10,
        maxValue: 22050,
        step: 1,
      },
      peak: {
        value: 17.18,
        minValue: 0.0001,
        maxValue: 20,
        step: 0.0001,
      },
    },
    createEffect: new Pizzicato.Effects.LowPassFilter({
      frequency: 12308,
      peak: 17.18,
    }),
    isVisible: true,
  },
  {
    name: 'High-Pass Filter',
    effects: {
      frequency: {
        value: 10,
        minValue: 10,
        maxValue: 22050,
        step: 1,
      },
      peak: {
        value: 0.0001,
        minValue: 0.0001,
        maxValue: 20,
        step: 0.0001,
      },
    },
    createEffect: new Pizzicato.Effects.HighPassFilter({
      frequency: 10,
      peak: 0.0001,
    }),
    isVisible: false,
  },
  {
    name: 'Ring Modulator',
    effects: {
      speed: {
        value: 0,
        minValue: 0,
        maxValue: 2000,
        step: 1,
      },
      distortion: {
        value: 0.2,
        minValue: 0.2,
        maxValue: 50,
        step: 0.0001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.RingModulator({
      speed: 0,
      distortion: 0.2,
      mix: 0,
    }),
    isVisible: false,
  },
];

export const ROCK_PRESET_ARRAY = [
  {
    name: 'Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.Delay({
      feedback: 0,
      time: 0,
      mix: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Ping-Pong Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.PingPongDelay(
      {
        feedback: 0,
        time: 0,
        mix: 0,
      },
    ),
    isVisible: false,
  },
  {
    name: 'Dub Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      cutoff: {
        value: 0,
        minValue: 0,
        maxValue: 4000,
        step: 100,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.DubDelay(
      {
        feedback: 0,
        time: 0,
        cutoff: 0,
        mix: 0,
      },
    ),
    isVisible: false,
  },
  {
    name: 'Distortion',
    effects: {
      gain: {
        value: 0.1,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Distortion({
      gain: 0.1,
    }),
    isVisible: true,
  },
  {
    name: 'Quadrafuzz',
    effects: {
      lowGain: {
        value: 1,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      midLowGain: {
        value: 1,
        minValue: 1,
        maxValue: 1,
        step: 0.01,
      },
      midHighGain: {
        value: 1,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      highGain: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 1,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Quadrafuzz({
      lowGain: 1,
      midLowGain: 1,
      midHighGain: 1,
      highGain: 0,
      mix: 1,
    }),
    isVisible: true,
  },
  {
    name: 'Flanger',
    effects: {
      time: {
        value: 0.3,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      speed: {
        value: 0.3,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      depth: {
        value: 0.2,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      feedback: {
        value: 0.15,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 0.09,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Flanger({
      time: 0.3,
      speed: 0.3,
      depth: 0.2,
      feedback: 0.15,
      mix: 0.09,
    }),
    isVisible: true,
  },
  {
    name: 'Reverb',
    effects: {
      time: {
        value: 1.39,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
      decay: {
        value: 2.66,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
      mix: {
        value: 0.53,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Reverb({
      time: 1.39,
      decay: 2.66,
      mix: 0.53,
    }),
    isVisible: true,
  },
  {
    name: 'Tremolo',
    effects: {
      speed: {
        value: 17,
        minValue: 0,
        maxValue: 20,
        step: 0.01,
      },
      depth: {
        value: 0.25,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 1,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Tremolo({
      speed: 17,
      depth: 0.25,
      mix: 1,
    }),
    isVisible: true,
  },
  {
    name: 'Stereo panner',
    effects: {
      pan: {
        value: 0,
        minValue: -1,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.StereoPanner({
      pan: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Compressor',
    effects: {
      threshold: {
        value: -20,
        minValue: -100,
        maxValue: 0,
        step: 1,
      },
      knee: {
        value: 10,
        minValue: 0,
        maxValue: 40,
        step: 1,
      },
      attack: {
        value: 0.85,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
      release: {
        value: 0.3,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
      ratio: {
        value: 14,
        minValue: 1,
        maxValue: 20,
        step: 1,
      },
    },
    createEffect: new Pizzicato.Effects.Compressor({
      threshold: -20,
      knee: 10,
      attack: 0.85,
      release: 0.3,
      ratio: 14,
    }),
    isVisible: true,
  },
  {
    name: 'Low-Pass Filter',
    effects: {
      frequency: {
        value: 22050,
        minValue: 10,
        maxValue: 22050,
        step: 1,
      },
      peak: {
        value: 0.0001,
        minValue: 0.0001,
        maxValue: 20,
        step: 0.0001,
      },
    },
    createEffect: new Pizzicato.Effects.LowPassFilter({
      frequency: 22050,
      peak: 0.0001,
    }),
    isVisible: false,
  },
  {
    name: 'High-Pass Filter',
    effects: {
      frequency: {
        value: 10,
        minValue: 10,
        maxValue: 22050,
        step: 1,
      },
      peak: {
        value: 0.0001,
        minValue: 0.0001,
        maxValue: 20,
        step: 0.0001,
      },
    },
    createEffect: new Pizzicato.Effects.HighPassFilter({
      frequency: 10,
      peak: 0.0001,
    }),
    isVisible: true,
  },
  {
    name: 'Ring Modulator',
    effects: {
      speed: {
        value: 0,
        minValue: 0,
        maxValue: 2000,
        step: 1,
      },
      distortion: {
        value: 0.2,
        minValue: 0.2,
        maxValue: 50,
        step: 0.0001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.RingModulator({
      speed: 0,
      distortion: 0.2,
      mix: 0,
    }),
    isVisible: false,
  },
];

export const POP_PRESET_ARRAY = [
  {
    name: 'Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.Delay({
      feedback: 0,
      time: 0,
      mix: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Ping-Pong Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.PingPongDelay(
      {
        feedback: 0,
        time: 0,
        mix: 0,
      },
    ),
    isVisible: false,
  },
  {
    name: 'Dub Delay',
    effects: {
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      time: {
        value: 0,
        minValue: 0,
        maxValue: 5,
        step: 0.001,
      },
      cutoff: {
        value: 0,
        minValue: 0,
        maxValue: 4000,
        step: 100,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
    },
    createEffect: new Pizzicato.Effects.DubDelay(
      {
        feedback: 0,
        time: 0,
        cutoff: 0,
        mix: 0,
      },
    ),
    isVisible: false,
  },
  {
    name: 'Distortion',
    effects: {
      gain: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Distortion({
      gain: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Quadrafuzz',
    effects: {
      lowGain: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      midLowGain: {
        value: 0.04,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      midHighGain: {
        value: 0.13,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      highGain: {
        value: 0.46,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 0.09,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Quadrafuzz({
      lowGain: 0,
      midLowGain: 0.04,
      midHighGain: 0.13,
      highGain: 0.46,
      mix: 0.09,
    }),
    isVisible: true,
  },
  {
    name: 'Flanger',
    effects: {
      time: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      speed: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      depth: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      feedback: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Flanger({
      time: 0,
      speed: 0,
      depth: 0,
      feedback: 0,
      mix: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Reverb',
    effects: {
      time: {
        value: 2.55,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
      decay: {
        value: 2.55,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
      mix: {
        value: 2.55,
        minValue: 0,
        maxValue: 3,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Reverb({
      time: 2.55,
      decay: 2.55,
      mix: 2.55,
    }),
    isVisible: true,
  },
  {
    name: 'Tremolo',
    effects: {
      speed: {
        value: 13.58,
        minValue: 0,
        maxValue: 20,
        step: 0.01,
      },
      depth: {
        value: 0.05,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
      mix: {
        value: 0.79,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.Tremolo({
      speed: 13.58,
      depth: 0.05,
      mix: 0.79,
    }),
    isVisible: true,
  },
  {
    name: 'Stereo panner',
    effects: {
      pan: {
        value: 0,
        minValue: -1,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.StereoPanner({
      pan: 0,
    }),
    isVisible: false,
  },
  {
    name: 'Compressor',
    effects: {
      threshold: {
        value: -47,
        minValue: -100,
        maxValue: 0,
        step: 1,
      },
      knee: {
        value: 32,
        minValue: 0,
        maxValue: 40,
        step: 1,
      },
      attack: {
        value: 0.82,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
      release: {
        value: 0.94,
        minValue: 0,
        maxValue: 1,
        step: 0.001,
      },
      ratio: {
        value: 18,
        minValue: 1,
        maxValue: 20,
        step: 1,
      },
    },
    createEffect: new Pizzicato.Effects.Compressor({
      threshold: -47,
      knee: 32,
      attack: 0.82,
      release: 0.94,
      ratio: 18,
    }),
    isVisible: true,
  },
  {
    name: 'Low-Pass Filter',
    effects: {
      frequency: {
        value: 6217,
        minValue: 10,
        maxValue: 22050,
        step: 1,
      },
      peak: {
        value: 9.97,
        minValue: 0.0001,
        maxValue: 20,
        step: 0.0001,
      },
    },
    createEffect: new Pizzicato.Effects.LowPassFilter({
      frequency: 6217,
      peak: 9.97,
    }),
    isVisible: true,
  },
  {
    name: 'High-Pass Filter',
    effects: {
      frequency: {
        value: 10,
        minValue: 10,
        maxValue: 22050,
        step: 1,
      },
      peak: {
        value: 0.0001,
        minValue: 0.0001,
        maxValue: 20,
        step: 0.0001,
      },
    },
    createEffect: new Pizzicato.Effects.HighPassFilter({
      frequency: 10,
      peak: 0.0001,
    }),
    isVisible: false,
  },
  {
    name: 'Ring Modulator',
    effects: {
      speed: {
        value: 0,
        minValue: 0,
        maxValue: 2000,
        step: 1,
      },
      distortion: {
        value: 0.2,
        minValue: 0.2,
        maxValue: 50,
        step: 0.0001,
      },
      mix: {
        value: 0,
        minValue: 0,
        maxValue: 1,
        step: 0.01,
      },
    },
    createEffect: new Pizzicato.Effects.RingModulator({
      speed: 0,
      distortion: 0.2,
      mix: 0,
    }),
    isVisible: false,
  },
];
