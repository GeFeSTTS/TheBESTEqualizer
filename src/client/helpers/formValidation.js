import {
  minLength,
  maxLength,
  errorMessages,
  regExpPatterns,
} from './validationHelpers';

export const formValidation = (formData) => {
  const errors = {};
  const patterns = regExpPatterns(formData.password);

  Object.keys(formData).forEach((fieldValue) => {
    if (formData[fieldValue]) {
      if (formData[fieldValue].length <= minLength[fieldValue]) {
        errors[fieldValue] = errorMessages(fieldValue, minLength[fieldValue]).toShort;
      } else if (formData[fieldValue].length >= maxLength[fieldValue]) {
        errors[fieldValue] = errorMessages(fieldValue, maxLength[fieldValue]).toLong;
      } else if (!patterns[fieldValue].test(formData[fieldValue])) {
        errors[fieldValue] = errorMessages(fieldValue).failedRegExp;
      }
    } else {
      errors[fieldValue] = errorMessages(fieldValue).emptyField;
    }
  });

  return errors;
};
