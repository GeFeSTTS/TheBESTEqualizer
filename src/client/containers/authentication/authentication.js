import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import FormComponent from '../../components/formComponent';
import { fieldsInfo } from '../../helpers/constants';
import { formValidation } from '../../helpers/formValidation';
import { postUserData, authFail } from '../../store/actions/postUserDataAction';
import authImage from '../../assets/images/authImage.png';
import './authentication.css';

export class Authentication extends Component {
    state = {
      isMember: true,
      userData: {},
      validationErrors: {},
    };

    static getDerivedStateFromProps(props) {
      if (props.error) {
        return {
          validationErrors: props.error,
        };
      }
      return null;
    }

    onInputChange = ({ target: { name, value } }) => {
      const userData = { ...this.state.userData };
      userData[name] = value;
      this.setState({ userData });
    };

    onLinkClick = () => {
      const { isMember } = this.state;
      this.setState({ isMember: !isMember, userData: {}, validationErrors: {} });
      this.props.authFailAction(null);
    };

    onFormSubmit = () => {
      const { history, onAuth } = this.props;
      const { data, path } = this.isMemberInfo();
      const validationRes = formValidation(data);

      if (Object.keys(validationRes).length === 0) {
        onAuth(path, data, history);
      } else {
        this.setState({ validationErrors: validationRes });
      }
    };

    isMemberInfo = (info = []) => {
      const {
        userData: {
          username, email, password, passwordConfirmation,
        },
        isMember,
      } = this.state;
      if (isMember) {
        return {
          fildsToRender: info.filter(el => isMember === el.isMember),
          formTitle: 'Login',
          message: 'Dont have an account? Register!',
          path: '/login',
          data: { email, password },
        };
      }
      return {
        fildsToRender: info,
        formTitle: 'Registration',
        message: 'Already have an account? Login!',
        path: '/registration',
        data: {
          username, email, password, passwordConfirmation,
        },
      };
    }

    render() {
      const { userData, validationErrors } = this.state;
      const { fildsToRender, formTitle, message } = this.isMemberInfo(fieldsInfo);
      return (
        <div className="authentication">
          <img
            type="image/svg+xml"
            src={authImage}
            alt="Music band"
            className="authImage"
          />
          <div className="form-container">
            <h1 className="title">{formTitle}</h1>
            <FormComponent
              fieldsToRender={fildsToRender}
              onInputChange={this.onInputChange}
              onFormSubmit={this.onFormSubmit}
              userData={userData}
              validationErrors={validationErrors}
              loading={this.props.loading}
            />
            <input
              type="button"
              onClick={this.onLinkClick}
              className="message"
              value={message}
            />
          </div>
        </div>
      );
    }
}

Authentication.propTypes = {
  history: PropTypes.instanceOf(Object).isRequired,
  onAuth: PropTypes.func,
  loading: PropTypes.bool,
  authFailAction: PropTypes.func,
};

const mapDispatchToProps = {
  onAuth: postUserData,
  authFailAction: authFail,
};

const mapStateToProps = state => ({
  loading: state.authStatus.loading,
  error: state.authStatus.error,
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Authentication));
