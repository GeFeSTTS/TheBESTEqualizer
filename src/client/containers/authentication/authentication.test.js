import checkPropTypes from 'check-prop-types';
import React from 'react';
import { shallow } from 'enzyme';
import { Authentication } from './authentication';
import { formValidation } from '../../helpers/formValidation';
import { fieldsInfo } from '../../helpers/constants';

jest.mock('../../helpers/constants', () => ({
  fieldsInfo: [
    {
      label: 'RegisterLabel',
      isMember: false,
    },
    {
      label: 'LoginLabel',
      isMember: true,
    }],
}));

jest.mock('../../helpers/formValidation');

const checkProps = (component, expectedProps) => {
  const propError = checkPropTypes(component.propTypes, expectedProps, 'prop', component.name);
  expect(propError).toBeUndefined();
};

const defaultProps = {
  authFailAction: jest.fn(),
  onAuth: jest.fn(),
  history: {},
  loading: false,
  error: null,
};

describe('render Auth', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Authentication {...defaultProps} />);
  });

  describe('render <Auth />', () => {
    it('should render Authentication properly', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('updation state with server validation errors', () => {
    it('Update state if there are server errors', () => {
      wrapper.setProps({ error: { email: 'server error' } });
      expect(wrapper.state().validationErrors).toEqual({ email: 'server error' });
    });
    it('Does not update state if there are no server errors', () => {
      wrapper.setProps({ error: {} });
      expect(wrapper.state().validationErrors).toEqual({});
    });
  });

  describe('test "onFormSubmit" functionality', () => {
    const testingOnFormSubmit = (fakeErrors) => {
      formValidation.mockReturnValue(fakeErrors);
      wrapper.update();
      wrapper.instance().onFormSubmit();
    };
    it('it should setState when "formValidation" did not pass', () => {
      const fakeErrors = { email: 'fake email is not valid' };
      testingOnFormSubmit(fakeErrors);
      expect(wrapper.instance().state.validationErrors).toEqual(fakeErrors);
    });

    it('it should call action when "formValidation" was passed', () => {
      const fakeErrors = {};
      testingOnFormSubmit(fakeErrors);
      expect(wrapper.instance().props.onAuth).toBeCalled();
    });
  });

  describe('test "onInputChange" functionality', () => {
    it('it should setState when "onInputChange" is called', () => {
      const fakeInputData = { email: 'fakeEmail' };
      wrapper.instance().onInputChange({ target: { name: 'email', value: 'fakeEmail' } });
      expect(wrapper.instance().state.userData).toEqual(fakeInputData);
    });
  });

  describe('test if user already has an account', () => {
    it('the initial state of "isMember" should be "true"', () => {
      expect(wrapper.state().isMember).toBe(true);
    });
    it('"onLinkClick" should toggle the state of "isMember', () => {
      const expectedState = !wrapper.state().isMember;
      wrapper.instance().onLinkClick();
      wrapper.update();
      expect(wrapper.state().isMember).toBe(expectedState);
      expect(wrapper.instance().props.error).toEqual(null);
    });
    it('should called "onLinkClick" function on "click" event', () => {
      const testClick = jest.spyOn(wrapper.instance(), 'onLinkClick');
      wrapper.instance().forceUpdate();
      wrapper.find('.message').simulate('click');
      expect(testClick).toHaveBeenCalled();
    });
  });

  describe('Fields info to render', () => {
    it('should render "Login" form first', () => {
      wrapper.instance().setState({ userData: { email: 'fakeEmail', password: 'fakePassword' } });
      const expectedResult = {
        fildsToRender: [{
          label: 'LoginLabel',
          isMember: true,
        }],
        formTitle: 'Login',
        message: 'Dont have an account? Register!',
        path: '/login',
        data: wrapper.instance().state.userData,
      };
      expect(wrapper.instance().isMemberInfo(fieldsInfo).fildsToRender.length).toBe(1);
      expect(wrapper.instance().isMemberInfo(fieldsInfo)).toEqual(expectedResult);
    });
    it('should render "Registration" form when user is not member yet', () => {
      wrapper.instance().setState({
        userData: {
          email: 'fakeEmail',
          password: 'fakePassword',
          username: 'fakeUser',
          passwordConfirmation: 'fakePassword',
        },
        isMember: false,
      });
      const expectedResult = {
        fildsToRender: fieldsInfo,
        formTitle: 'Registration',
        message: 'Already have an account? Login!',
        path: '/registration',
        data: wrapper.instance().state.userData,
      };
      expect(wrapper.instance().isMemberInfo(fieldsInfo).fildsToRender.length).toBe(2);
      expect(wrapper.instance().isMemberInfo(fieldsInfo)).toEqual(expectedResult);
    });
  });

  describe('testing ptopTypes', () => {
    it('does not throw warning with expected props', () => {
      checkProps(Authentication, defaultProps);
    });
  });
});
