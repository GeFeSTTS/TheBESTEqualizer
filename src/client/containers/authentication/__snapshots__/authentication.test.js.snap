// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Auth render <Auth /> should render Authentication properly 1`] = `
"<div className=\\"authentication\\">
  <img type=\\"image/svg+xml\\" src=\\"test-file-stub\\" alt=\\"Music band\\" className=\\"authImage\\" />
  <div className=\\"form-container\\">
    <h1 className=\\"title\\">
      Login
    </h1>
    <FormComponent fieldsToRender={{...}} onInputChange={[Function]} onFormSubmit={[Function]} userData={{...}} validationErrors={{...}} loading={false} />
    <input type=\\"button\\" onClick={[Function]} className=\\"message\\" value=\\"Dont have an account? Register!\\" />
  </div>
</div>"
`;
